In [1]:
# ライブラリのインポート
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
# 元となるデータの取得
df_origin = pd.read_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-9/user-9-nonull.csv")
df_origin.head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
0           0                    0          2267020723          NaN   
1           1                18909          2267039632  Eye Tracker   
2           2                22084          2267042807        Mouse   
3           3                51052          2267071775  Eye Tracker   
4           4                67696          2267088419  Eye Tracker   

  Project name Export date Participant name Recording name Recording date  \
0          JD4  2025/07/18           user-9    Recording10     2025/07/18   
1          JD4  2025/07/18           user-9    Recording10     2025/07/18   
2          JD4  2025/07/18           user-9    Recording10     2025/07/18   
3          JD4  2025/07/18           user-9    Recording10     2025/07/18   
4          JD4  2025/07/18           user-9    Recording10     2025/07/18   

  Recording date UTC  ... Original Media height Eye movement type  \
0         2025/07/18  ...                   NaN               NaN   
1         2025/07/18  ...                   NaN      EyesNotFound   
2         2025/07/18  ...                   NaN      EyesNotFound   
3         2025/07/18  ...                   NaN      EyesNotFound   
4         2025/07/18  ...                   NaN      EyesNotFound   

   Eye movement event duration Eye movement type index Fixation point X  \
0                          NaN                     NaN              NaN   
1                        115.0                     1.0              NaN   
2                        115.0                     1.0              NaN   
3                        115.0                     1.0              NaN   
4                        115.0                     1.0              NaN   

  Fixation point Y  Fixation point X (MCSnorm)  Fixation point Y (MCSnorm)  \
0              NaN                         NaN                         NaN   
1              NaN                         NaN                         NaN   
2              NaN                         NaN                         NaN   
3              NaN                         NaN                         NaN   
4              NaN                         NaN                         NaN   

   Mouse position X  Mouse position Y  
0               NaN               NaN  
1               NaN               NaN  
2            1851.0             947.0  
3               NaN               NaN  
4               NaN               NaN  

[5 rows x 84 columns]

# カラムの整理

In [3]:
nunique_series = df_origin.nunique(dropna=False).to_string()
print(nunique_series)

Unnamed: 0                                    13593
Recording timestamp                           13580
Computer timestamp                            13580
Sensor                                            3
Project name                                      1
Export date                                       1
Participant name                                  1
Recording name                                    1
Recording date                                    1
Recording date UTC                                1
Recording start time                              1
Recording start time UTC                          1
Recording duration                                1
Timeline name                                     1
Recording Fixation filter name                    1
Recording software version                        1
Recording resolution height                       1
Recording resolution width                        1
Recording monitor latency                         1
Average cali

In [4]:
# 欠損値も含めてユニークな値の個数をカウント
nunique_series = df_origin.nunique(dropna=False)

# ユニークな値が1つのカラム名リスト
single_unique_cols = nunique_series[nunique_series == 1].index.tolist()

In [5]:
# カラム名とそのユニーク値を出力
for col in single_unique_cols:
    unique_val = df_origin[col].unique()
    print(f"{col}: {unique_val}")


Project name: ['JD4']
Export date: ['2025/07/18']
Participant name: ['user-9']
Recording name: ['Recording10']
Recording date: ['2025/07/18']
Recording date UTC: ['2025/07/18']
Recording start time: ['17:06:44.023']
Recording start time UTC: ['08:06:44.023']
Recording duration: [243039]
Timeline name: ['Timeline1']
Recording Fixation filter name: ['Tobii I-VT (Fixation)']
Recording software version: ['24.21.435']
Recording resolution height: [1080]
Recording resolution width: [1920]
Recording monitor latency: [10]
Average calibration accuracy (mm): [49.3]
Average calibration accuracy (degrees): [3.61]
Average calibration accuracy (pixels): [171]
Average validation accuracy (mm): [13]
Average validation precision SD (mm): [17.4]
Average validation precision RMS (mm): [18]
Average validation accuracy (degrees): [1.2]
Average validation precision SD (degrees): [1.6]
Average validation precision RMS (degrees): [1.65]
Average validation accuracy (pixels): [45]
Average validation precision S

## カラムの要素が1つしかない変数の説明
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-9`|
|Recording name| 録画名|`Recording10`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`17:06:44.023`|
|Recording start time UTC| 録画開始時刻(UTC) |`08:06:44.023`|
|Recording duration| 録画時間|`243039`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`49.3`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`3.61`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`171`pixels|
|Average validation accuracy | 平均バリデーション精度|`13`mm|
|Average validation precision SD | 平均バリデーション予測SD|`17.4`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`18`mm|
|Average validation accuracy |平均バリデーション精度| `1.2`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `1.6`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `1.65`degrees|
|Average validation accuracy |平均バリデーション精度| `45`pixels|
|Average validation precision SD |平均バリデーション予測SD| `61`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `63`pixels|


In [6]:
# 削除した新しいデータフレーム
df_cleaned = df_origin.drop(columns=single_unique_cols)
df_cleaned.head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
0           0                    0          2267020723          NaN   
1           1                18909          2267039632  Eye Tracker   
2           2                22084          2267042807        Mouse   
3           3                51052          2267071775  Eye Tracker   
4           4                67696          2267088419  Eye Tracker   

   Eyetracker timestamp           Event Event value  Gaze point X  \
0                   NaN  RecordingStart         NaN           NaN   
1          2.282300e+09             NaN         NaN           NaN   
2                   NaN             NaN         NaN           NaN   
3          2.282332e+09             NaN         NaN           NaN   
4          2.282349e+09             NaN         NaN           NaN   

   Gaze point Y  Gaze point left X  ...  Original Media height  \
0           NaN                NaN  ...                    NaN   
1           NaN                NaN  ...                    NaN   
2           NaN                NaN  ...                    NaN   
3           NaN                NaN  ...                    NaN   
4           NaN                NaN  ...                    NaN   

   Eye movement type  Eye movement event duration  Eye movement type index  \
0                NaN                          NaN                      NaN   
1       EyesNotFound                        115.0                      1.0   
2       EyesNotFound                        115.0                      1.0   
3       EyesNotFound                        115.0                      1.0   
4       EyesNotFound                        115.0                      1.0   

   Fixation point X  Fixation point Y  Fixation point X (MCSnorm)  \
0               NaN               NaN                         NaN   
1               NaN               NaN                         NaN   
2               NaN               NaN                         NaN   
3               NaN               NaN                         NaN   
4               NaN               NaN                         NaN   

   Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
0                         NaN               NaN               NaN  
1                         NaN               NaN               NaN  
2                         NaN            1851.0             947.0  
3                         NaN               NaN               NaN  
4                         NaN               NaN               NaN  

[5 rows x 57 columns]

## ユニークな値を取り除いたデータフレーム

In [7]:
print(df_cleaned.nunique(dropna=False).to_string())

Unnamed: 0                             13593
Recording timestamp                    13580
Computer timestamp                     13580
Sensor                                     3
Eyetracker timestamp                   13337
Event                                      9
Event value                                9
Gaze point X                            1786
Gaze point Y                             924
Gaze point left X                       1789
Gaze point left Y                        915
Gaze point right X                      1757
Gaze point right Y                       937
Gaze direction left X                   9919
Gaze direction left Y                   9200
Gaze direction left Z                   5649
Gaze direction right X                  9784
Gaze direction right Y                  9199
Gaze direction right Z                  5770
Pupil diameter left                      948
Pupil diameter right                     974
Pupil diameter filtered                  941
Validity l

## Sensor事にデータフレームを分割

In [8]:
# NaNを含む行
df_sensor_nan = df_cleaned[df_cleaned['Sensor'].isna()]

# "Mouse" の行
df_sensor_mouse = df_cleaned[df_cleaned['Sensor'] == "Mouse"]

# "Eye Tracker" の行
df_sensor_eye = df_cleaned[df_cleaned['Sensor'] == "Eye Tracker"]


### 問題の既知代わりの時間を取得する

In [9]:
df_sensor_nan

Unnamed: 0  Recording timestamp  Computer timestamp Sensor  \
0               0                    0          2267020723    NaN   
259           259              2840826          2269861549    NaN   
485           485              5860724          2272881447    NaN   
493           493              5942400          2272963123    NaN   
1799         1799             44200635          2311221358    NaN   
1807         1807             44380684          2311401407    NaN   
1815         1815             44729779          2311750502    NaN   
1816         1816             44729779          2311750502    NaN   
2103         2103             49755162          2316775885    NaN   
2104         2104             49755162          2316775885    NaN   
4849         4849             95874639          2362895362    NaN   
4852         4852             95909592          2362930315    NaN   
4853         4853             95909592          2362930315    NaN   
5134         5134            100900781          2367921504    NaN   
5135         5135            100900781          2367921504    NaN   
8000         8000            148779816          2415800539    NaN   
8002         8002            148809620          2415830343    NaN   
8003         8003            148809620          2415830343    NaN   
8294         8294            153817632          2420838355    NaN   
8295         8295            153817632          2420838355    NaN   
10531       10531            191052751          2458073474    NaN   
10533       10533            191084239          2458104962    NaN   
10534       10534            191084239          2458104962    NaN   
10811       10811            196091507          2463112230    NaN   
10812       10812            196091507          2463112230    NaN   
13319       13319            237819190          2504839913    NaN   
13322       13322            237854039          2504874762    NaN   
13323       13323            237854039          2504874762    NaN   
13589       13589            242845015          2509865738    NaN   
13592       13592            243038771          2510059494    NaN   

       Eyetracker timestamp                          Event Event value  \
0                       NaN                 RecordingStart         NaN   
259                     NaN  Eye tracker Calibration start         NaN   
485                     NaN                     MouseEvent  Down, Left   
493                     NaN                     MouseEvent    Up, Left   
1799                    NaN                     MouseEvent  Down, Left   
1807                    NaN                     MouseEvent    Up, Left   
1815                    NaN    Eye tracker Calibration end         NaN   
1816                    NaN             ImageStimulusStart      blank2   
2103                    NaN               ImageStimulusEnd      blank2   
2104                    NaN             ImageStimulusStart  視線誘導問題.002   
4849                    NaN                  KeyboardEvent       Space   
4852                    NaN               ImageStimulusEnd  視線誘導問題.002   
4853                    NaN             ImageStimulusStart      blank2   
5134                    NaN               ImageStimulusEnd      blank2   
5135                    NaN             ImageStimulusStart  視線誘導問題.001   
8000                    NaN                  KeyboardEvent       Space   
8002                    NaN               ImageStimulusEnd  視線誘導問題.001   
8003                    NaN             ImageStimulusStart      blank2   
8294                    NaN               ImageStimulusEnd      blank2   
8295                    NaN             ImageStimulusStart  視線誘導問題.003   
10531                   NaN                  KeyboardEvent       Space   
10533                   NaN               ImageStimulusEnd  視線誘導問題.003   
10534                   NaN             ImageStimulusStart      blank2   
10811                   NaN               ImageStimulusEnd      blank2   
10812            

In [10]:
# "視線誘導問題." を含む行を抽出
df_filtered = df_sensor_nan[df_sensor_nan["Event value"].astype(str).str.contains("視線誘導問題.")].copy()

# 結果の確認（先頭5行を表示）
df_filtered.loc[:,['Recording timestamp', 'Computer timestamp', 'Event value']]
print(df_filtered.loc[:,['Recording timestamp', 'Computer timestamp', 'Event value']])

       Recording timestamp  Computer timestamp Event value
2104              49755162          2316775885  視線誘導問題.002
4852              95909592          2362930315  視線誘導問題.002
5135             100900781          2367921504  視線誘導問題.001
8002             148809620          2415830343  視線誘導問題.001
8295             153817632          2420838355  視線誘導問題.003
10533            191084239          2458104962  視線誘導問題.003
10812            196091507          2463112230  視線誘導問題.004
13322            237854039          2504874762  視線誘導問題.004


### 各問題の開始時刻
|       |Recording timestamp|  Computer timestamp| Event value|
|-------|-------------------|--------------------|------------|
|2104|49755162|2316775885|視線誘導問題.002|
|4852|95909592|2362930315|視線誘導問題.002|
|5135|100900781|2367921504|視線誘導問題.001|
|8002|148809620|2415830343|視線誘導問題.001|
|8295|153817632|2420838355|視線誘導問題.003|
|10533|191084239|2458104962|視線誘導問題.003|
|10812|196091507|2463112230|視線誘導問題.004|
|13322|237854039|2504874762|視線誘導問題.004|

In [11]:
df_filtered.loc[:,['Recording timestamp', 'Computer timestamp', 'Event value']].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-9/start-stop-timing.csv")

In [12]:
df_sensor_mouse

Unnamed: 0  Recording timestamp  Computer timestamp Sensor  \
2              2                22084          2267042807  Mouse   
5              5                77726          2267098449  Mouse   
8              8               101771          2267122494  Mouse   
11            11               139738          2267160461  Mouse   
13            13               155734          2267176457  Mouse   
...          ...                  ...                 ...    ...   
1831        1831             44886611          2311907334  Mouse   
1833        1833             44902548          2311923271  Mouse   
1835        1835             44919093          2311939816  Mouse   
1837        1837             44934592          2311955315  Mouse   
1838        1838             44944572          2311965295  Mouse   

      Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
2                      NaN   NaN         NaN           NaN           NaN   
5                      NaN   NaN         NaN           NaN           NaN   
8                      NaN   NaN         NaN           NaN           NaN   
11                     NaN   NaN         NaN           NaN           NaN   
13                     NaN   NaN         NaN           NaN           NaN   
...                    ...   ...         ...           ...           ...   
1831                   NaN   NaN         NaN           NaN           NaN   
1833                   NaN   NaN         NaN           NaN           NaN   
1835                   NaN   NaN         NaN           NaN           NaN   
1837                   NaN   NaN         NaN           NaN           NaN   
1838                   NaN   NaN         NaN           NaN           NaN   

      Gaze point left X  ...  Original Media height  Eye movement type  \
2                   NaN  ...                    NaN       EyesNotFound   
5                   NaN  ...                    NaN       EyesNotFound   
8                   NaN  ...                    NaN       EyesNotFound   
11                  NaN  ...                    NaN            Saccade   
13                  NaN  ...                    NaN            Saccade   
...                 ...  ...                    ...                ...   
1831                NaN  ...                 1080.0           Fixation   
1833                NaN  ...                 1080.0           Fixation   
1835                NaN  ...                 1080.0           Fixation   
1837                NaN  ...                 1080.0           Fixation   
1838                NaN  ...                 1080.0            Saccade   

      Eye movement event duration  Eye movement type index  Fixation point X  \
2                           115.0                      1.0               NaN   
5                           115.0                      1.0               NaN   
8                           115.0                      1.0               NaN   
11                           67.0                      1.0               NaN   
13                           67.0                      1.0               NaN   
...                           ...                      ...               ...   
1831                         83.0                     84.0            1008.0   
1833                         83.0                     84.0            1008.0   
1835                         83.0                     84.0            1008.0   
1837                         83.0                     84.0            1008.0   
1838                         16.0                    117.0               NaN   

      Fixation point Y  Fixation point X (MCSnorm)  \
2                  NaN                         NaN   
5                  NaN                         NaN   
8                  NaN                         NaN   
11                 NaN                         NaN   
13                 NaN                         NaN   
...                ...                         ...   
1831             581.0                      0.52

In [13]:
df_sensor_eye

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
1               1                18909          2267039632  Eye Tracker   
3               3                51052          2267071775  Eye Tracker   
4               4                67696          2267088419  Eye Tracker   
6               6                84383          2267105106  Eye Tracker   
7               7               101180          2267121903  Eye Tracker   
...           ...                  ...                 ...          ...   
13586       13586            242784265          2509804988  Eye Tracker   
13587       13587            242815284          2509836007  Eye Tracker   
13588       13588            242831561          2509852284  Eye Tracker   
13590       13590            242848248          2509868971  Eye Tracker   
13591       13591            242872178          2509892901  Eye Tracker   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
1              2.282300e+09   NaN         NaN           NaN           NaN   
3              2.282332e+09   NaN         NaN           NaN           NaN   
4              2.282349e+09   NaN         NaN           NaN           NaN   
6              2.282365e+09   NaN         NaN           NaN           NaN   
7              2.282382e+09   NaN         NaN           NaN           NaN   
...                     ...   ...         ...           ...           ...   
13586          2.525050e+09   NaN         NaN           NaN           NaN   
13587          2.525081e+09   NaN         NaN           NaN           NaN   
13588          2.525098e+09   NaN         NaN           NaN           NaN   
13590          2.525114e+09   NaN         NaN           NaN           NaN   
13591          2.525138e+09   NaN         NaN           NaN           NaN   

       Gaze point left X  ...  Original Media height  Eye movement type  \
1                    NaN  ...                    NaN       EyesNotFound   
3                    NaN  ...                    NaN       EyesNotFound   
4                    NaN  ...                    NaN       EyesNotFound   
6                    NaN  ...                    NaN       EyesNotFound   
7                    NaN  ...                    NaN       EyesNotFound   
...                  ...  ...                    ...                ...   
13586                NaN  ...                 1080.0       EyesNotFound   
13587                NaN  ...                 1080.0       EyesNotFound   
13588                NaN  ...                 1080.0       EyesNotFound   
13590                NaN  ...                    NaN       EyesNotFound   
13591                NaN  ...                    NaN       EyesNotFound   

       Eye movement event duration  Eye movement type index  Fixation point X  \
1                            115.0                      1.0               NaN   
3                            115.0                      1.0               NaN   
4                            115.0                      1.0               NaN   
6                            115.0                      1.0               NaN   
7                            115.0                      1.0               NaN   
...                            ...                      ...               ...   
13586                        732.0                    253.0               NaN   
13587                        732.0                    253.0               NaN   
13588                        732.0                    253.0               NaN   
13590                        732.0                    253.0               NaN   
13591                        732.0                    253.0               NaN   

       Fixation point Y  Fixation point X (MCSnorm)  \
1                   NaN                         NaN   
3                   NaN                         NaN   
4                   NaN                         NaN   
6                   NaN                         NaN   
7                   NaN         

## Mouse以外のデータを抽出

In [14]:
df_cleaned_ver2 = df_cleaned[df_cleaned['Sensor'] != "Mouse"]
nunique_series_ver2 = df_cleaned_ver2.nunique(dropna=False)
single_unique_cols_ver2 = nunique_series_ver2[nunique_series_ver2 == 1].index.tolist()
# カラム名とそのユニーク値を出力
for col in single_unique_cols_ver2:
    unique_val = df_cleaned_ver2[col].unique()
    print(f"{col}: {unique_val}")

Mouse position X: [nan]
Mouse position Y: [nan]


In [15]:
df_cleaned_ver2 = df_cleaned_ver2.drop(columns=single_unique_cols_ver2)

In [16]:
print(df_cleaned_ver2.nunique(dropna=False).to_string())

Unnamed: 0                             13366
Recording timestamp                    13357
Computer timestamp                     13357
Sensor                                     2
Eyetracker timestamp                   13337
Event                                      9
Event value                                9
Gaze point X                            1786
Gaze point Y                             924
Gaze point left X                       1789
Gaze point left Y                        915
Gaze point right X                      1757
Gaze point right Y                       937
Gaze direction left X                   9919
Gaze direction left Y                   9200
Gaze direction left Z                   5649
Gaze direction right X                  9784
Gaze direction right Y                  9199
Gaze direction right Z                  5770
Pupil diameter left                      948
Pupil diameter right                     974
Pupil diameter filtered                  941
Validity l

In [17]:
df_cleaned_ver2.head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
0           0                    0          2267020723          NaN   
1           1                18909          2267039632  Eye Tracker   
3           3                51052          2267071775  Eye Tracker   
4           4                67696          2267088419  Eye Tracker   
6           6                84383          2267105106  Eye Tracker   

   Eyetracker timestamp           Event Event value  Gaze point X  \
0                   NaN  RecordingStart         NaN           NaN   
1          2.282300e+09             NaN         NaN           NaN   
3          2.282332e+09             NaN         NaN           NaN   
4          2.282349e+09             NaN         NaN           NaN   
6          2.282365e+09             NaN         NaN           NaN   

   Gaze point Y  Gaze point left X  ...  Presented Media position Y (DACSpx)  \
0           NaN                NaN  ...                                  NaN   
1           NaN                NaN  ...                                  NaN   
3           NaN                NaN  ...                                  NaN   
4           NaN                NaN  ...                                  NaN   
6           NaN                NaN  ...                                  NaN   

   Original Media width  Original Media height  Eye movement type  \
0                   NaN                    NaN                NaN   
1                   NaN                    NaN       EyesNotFound   
3                   NaN                    NaN       EyesNotFound   
4                   NaN                    NaN       EyesNotFound   
6                   NaN                    NaN       EyesNotFound   

   Eye movement event duration  Eye movement type index  Fixation point X  \
0                          NaN                      NaN               NaN   
1                        115.0                      1.0               NaN   
3                        115.0                      1.0               NaN   
4                        115.0                      1.0               NaN   
6                        115.0                      1.0               NaN   

   Fixation point Y  Fixation point X (MCSnorm)  Fixation point Y (MCSnorm)  
0               NaN                         NaN                         NaN  
1               NaN                         NaN                         NaN  
3               NaN                         NaN                         NaN  
4               NaN                         NaN                         NaN  
6               NaN                         NaN                         NaN  

[5 rows x 55 columns]

In [18]:
# 欠損値も含めてユニークな値の個数をカウント
nunique_series_ver3 = df_cleaned_ver2.nunique(dropna=False)

# ユニークな値が1つのカラム名リスト
single_unique_cols_ver3 = nunique_series_ver3[nunique_series_ver3 == 2].index.tolist()

# カラム名とそのユニーク値を出力
for col in single_unique_cols_ver3:
    unique_val = df_cleaned_ver2[col].unique()
    print(f"{col}: {unique_val}")


Sensor: [nan 'Eye Tracker']
Presented Media width: [  nan 1920.]
Presented Media height: [  nan 1080.]
Presented Media position X (DACSpx): [nan  0.]
Presented Media position Y (DACSpx): [nan  0.]
Original Media width: [  nan 1920.]
Original Media height: [  nan 1080.]


### Sensor=Mouse以外に着目した後に要素が少ない特徴量について
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Mouse position X| マウス位置 X|`nan`|
|Mouse position Y| マウス位置 Y|`nan`|
|Sensor| センサー|`nan`, `Eye Tracker`|
|Presented Media width|提示されたメディアの幅|`nan` `1920`|
|Presented Media height|提示されたメディアの高さ|`nan`, `1080`|
|Presented Media position X (DACSpx)| 提示されたメディア位置 X|`nan`,  `0`|
|Presented Media position Y (DACSpx)| 提示されたメディアの位置 Y|`nan`,  `0`|
|Original Media width| オリジナルメディアの幅|`nan`, `1920`|
|Original Media height| オリジナルメディアの高さ|`nan`,`1080`|

In [19]:
df_sensor_eye

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
1               1                18909          2267039632  Eye Tracker   
3               3                51052          2267071775  Eye Tracker   
4               4                67696          2267088419  Eye Tracker   
6               6                84383          2267105106  Eye Tracker   
7               7               101180          2267121903  Eye Tracker   
...           ...                  ...                 ...          ...   
13586       13586            242784265          2509804988  Eye Tracker   
13587       13587            242815284          2509836007  Eye Tracker   
13588       13588            242831561          2509852284  Eye Tracker   
13590       13590            242848248          2509868971  Eye Tracker   
13591       13591            242872178          2509892901  Eye Tracker   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
1              2.282300e+09   NaN         NaN           NaN           NaN   
3              2.282332e+09   NaN         NaN           NaN           NaN   
4              2.282349e+09   NaN         NaN           NaN           NaN   
6              2.282365e+09   NaN         NaN           NaN           NaN   
7              2.282382e+09   NaN         NaN           NaN           NaN   
...                     ...   ...         ...           ...           ...   
13586          2.525050e+09   NaN         NaN           NaN           NaN   
13587          2.525081e+09   NaN         NaN           NaN           NaN   
13588          2.525098e+09   NaN         NaN           NaN           NaN   
13590          2.525114e+09   NaN         NaN           NaN           NaN   
13591          2.525138e+09   NaN         NaN           NaN           NaN   

       Gaze point left X  ...  Original Media height  Eye movement type  \
1                    NaN  ...                    NaN       EyesNotFound   
3                    NaN  ...                    NaN       EyesNotFound   
4                    NaN  ...                    NaN       EyesNotFound   
6                    NaN  ...                    NaN       EyesNotFound   
7                    NaN  ...                    NaN       EyesNotFound   
...                  ...  ...                    ...                ...   
13586                NaN  ...                 1080.0       EyesNotFound   
13587                NaN  ...                 1080.0       EyesNotFound   
13588                NaN  ...                 1080.0       EyesNotFound   
13590                NaN  ...                    NaN       EyesNotFound   
13591                NaN  ...                    NaN       EyesNotFound   

       Eye movement event duration  Eye movement type index  Fixation point X  \
1                            115.0                      1.0               NaN   
3                            115.0                      1.0               NaN   
4                            115.0                      1.0               NaN   
6                            115.0                      1.0               NaN   
7                            115.0                      1.0               NaN   
...                            ...                      ...               ...   
13586                        732.0                    253.0               NaN   
13587                        732.0                    253.0               NaN   
13588                        732.0                    253.0               NaN   
13590                        732.0                    253.0               NaN   
13591                        732.0                    253.0               NaN   

       Fixation point Y  Fixation point X (MCSnorm)  \
1                   NaN                         NaN   
3                   NaN                         NaN   
4                   NaN                         NaN   
6                   NaN                         NaN   
7                   NaN         

## 各問題毎のデータへ切り分け

In [20]:
# 結果を辞書に格納
extracted_data = {}

targets = df_filtered["Event value"].unique()
for target in targets:
    timestamps = df_filtered[df_filtered["Event value"] == target]["Recording timestamp"].iloc[0:2]
    if len(timestamps) < 2:
        continue  # start/endが不足している場合スキップ

    start, end = timestamps.iloc[0], timestamps.iloc[1]

    extracted = df_sensor_eye[
        (df_sensor_eye["Recording timestamp"] >= start) &
        (df_sensor_eye["Recording timestamp"] <= end)
    ].copy()

    extracted_data[target] = extracted  # 日本語のキーOK

In [21]:
extracted_data["視線誘導問題.002"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
2105        2105             49757342          2316778065  Eye Tracker   
2106        2106             49773405          2316794128  Eye Tracker   
2107        2107             49838223          2316858946  Eye Tracker   
2108        2108             49869390          2316890113  Eye Tracker   
2109        2109             49885479          2316906202  Eye Tracker   

      Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
2105          2.332035e+09   NaN         NaN           NaN           NaN   
2106          2.332051e+09   NaN         NaN           NaN           NaN   
2107          2.332116e+09   NaN         NaN           NaN           NaN   
2108          2.332147e+09   NaN         NaN           NaN           NaN   
2109          2.332163e+09   NaN         NaN           NaN           NaN   

      Gaze point left X  ...  Original Media height  Eye movement type  \
2105                NaN  ...                 1080.0       EyesNotFound   
2106                NaN  ...                 1080.0       EyesNotFound   
2107                NaN  ...                 1080.0       EyesNotFound   
2108                NaN  ...                 1080.0       EyesNotFound   
2109                NaN  ...                 1080.0       EyesNotFound   

      Eye movement event duration  Eye movement type index  Fixation point X  \
2105                        567.0                     54.0               NaN   
2106                        567.0                     54.0               NaN   
2107                        567.0                     54.0               NaN   
2108                        567.0                     54.0               NaN   
2109                        567.0                     54.0               NaN   

      Fixation point Y  Fixation point X (MCSnorm)  \
2105               NaN                         NaN   
2106               NaN                         NaN   
2107               NaN                         NaN   
2108               NaN                         NaN   
2109               NaN                         NaN   

      Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
2105                         NaN               NaN               NaN  
2106                         NaN               NaN               NaN  
2107                         NaN               NaN               NaN  
2108                         NaN               NaN               NaN  
2109                         NaN               NaN               NaN  

[5 rows x 57 columns]

In [22]:
print(extracted_data["視線誘導問題.002"].nunique(dropna=False).to_string())

Unnamed: 0                             2746
Recording timestamp                    2746
Computer timestamp                     2746
Sensor                                    1
Eyetracker timestamp                   2746
Event                                     1
Event value                               1
Gaze point X                           1058
Gaze point Y                            691
Gaze point left X                      1093
Gaze point left Y                       695
Gaze point right X                     1068
Gaze point right Y                      708
Gaze direction left X                  2375
Gaze direction left Y                  2321
Gaze direction left Z                  2013
Gaze direction right X                 2315
Gaze direction right Y                 2263
Gaze direction right Z                 1939
Pupil diameter left                     639
Pupil diameter right                    644
Pupil diameter filtered                 633
Validity left                   

In [23]:
extracted_data["視線誘導問題.001"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
5136        5136            100915117          2367935840  Eye Tracker   
5137        5137            100931519          2367952242  Eye Tracker   
5138        5138            100947999          2367968722  Eye Tracker   
5139        5139            100965031          2367985754  Eye Tracker   
5140        5140            100981382          2368002105  Eye Tracker   

      Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
5136          2.383190e+09   NaN         NaN         816.0         659.0   
5137          2.383206e+09   NaN         NaN         818.0         670.0   
5138          2.383223e+09   NaN         NaN           NaN           NaN   
5139          2.383240e+09   NaN         NaN         792.0         628.0   
5140          2.383256e+09   NaN         NaN         800.0         597.0   

      Gaze point left X  ...  Original Media height  Eye movement type  \
5136                NaN  ...                 1080.0       Unclassified   
5137                NaN  ...                 1080.0       Unclassified   
5138                NaN  ...                 1080.0       EyesNotFound   
5139              796.0  ...                 1080.0       Unclassified   
5140              799.0  ...                 1080.0       Unclassified   

      Eye movement event duration  Eye movement type index  Fixation point X  \
5136                         33.0                     82.0               NaN   
5137                         33.0                     82.0               NaN   
5138                         17.0                    110.0               NaN   
5139                         33.0                     83.0               NaN   
5140                         33.0                     83.0               NaN   

      Fixation point Y  Fixation point X (MCSnorm)  \
5136               NaN                         NaN   
5137               NaN                         NaN   
5138               NaN                         NaN   
5139               NaN                         NaN   
5140               NaN                         NaN   

      Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
5136                         NaN               NaN               NaN  
5137                         NaN               NaN               NaN  
5138                         NaN               NaN               NaN  
5139                         NaN               NaN               NaN  
5140                         NaN               NaN               NaN  

[5 rows x 57 columns]

In [24]:
extracted_data["視線誘導問題.003"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
8296        8296            153824319          2420845042  Eye Tracker   
8297        8297            153840868          2420861591  Eye Tracker   
8298        8298            153857417          2420878140  Eye Tracker   
8299        8299            153874233          2420894956  Eye Tracker   
8300        8300            153890786          2420911509  Eye Tracker   

      Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
8296          2.436096e+09   NaN         NaN           NaN           NaN   
8297          2.436112e+09   NaN         NaN           NaN           NaN   
8298          2.436129e+09   NaN         NaN           NaN           NaN   
8299          2.436146e+09   NaN         NaN           NaN           NaN   
8300          2.436162e+09   NaN         NaN           NaN           NaN   

      Gaze point left X  ...  Original Media height  Eye movement type  \
8296                NaN  ...                 1080.0       EyesNotFound   
8297                NaN  ...                 1080.0       EyesNotFound   
8298                NaN  ...                 1080.0       EyesNotFound   
8299                NaN  ...                 1080.0       EyesNotFound   
8300                NaN  ...                 1080.0       EyesNotFound   

      Eye movement event duration  Eye movement type index  Fixation point X  \
8296                        183.0                    183.0               NaN   
8297                        183.0                    183.0               NaN   
8298                        183.0                    183.0               NaN   
8299                        183.0                    183.0               NaN   
8300                        183.0                    183.0               NaN   

      Fixation point Y  Fixation point X (MCSnorm)  \
8296               NaN                         NaN   
8297               NaN                         NaN   
8298               NaN                         NaN   
8299               NaN                         NaN   
8300               NaN                         NaN   

      Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
8296                         NaN               NaN               NaN  
8297                         NaN               NaN               NaN  
8298                         NaN               NaN               NaN  
8299                         NaN               NaN               NaN  
8300                         NaN               NaN               NaN  

[5 rows x 57 columns]

In [25]:
extracted_data["視線誘導問題.004"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
10813       10813            196093920          2463114643  Eye Tracker   
10814       10814            196110732          2463131455  Eye Tracker   
10815       10815            196127268          2463147991  Eye Tracker   
10816       10816            196143936          2463164659  Eye Tracker   
10817       10817            196160655          2463181378  Eye Tracker   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
10813          2.478363e+09   NaN         NaN           NaN           NaN   
10814          2.478380e+09   NaN         NaN           NaN           NaN   
10815          2.478396e+09   NaN         NaN           NaN           NaN   
10816          2.478413e+09   NaN         NaN           NaN           NaN   
10817          2.478430e+09   NaN         NaN           NaN           NaN   

       Gaze point left X  ...  Original Media height  Eye movement type  \
10813                NaN  ...                 1080.0       EyesNotFound   
10814                NaN  ...                 1080.0       EyesNotFound   
10815                NaN  ...                 1080.0       EyesNotFound   
10816                NaN  ...                 1080.0       EyesNotFound   
10817                NaN  ...                 1080.0       EyesNotFound   

       Eye movement event duration  Eye movement type index  Fixation point X  \
10813                        183.0                    225.0               NaN   
10814                        183.0                    225.0               NaN   
10815                        183.0                    225.0               NaN   
10816                        183.0                    225.0               NaN   
10817                        183.0                    225.0               NaN   

       Fixation point Y  Fixation point X (MCSnorm)  \
10813               NaN                         NaN   
10814               NaN                         NaN   
10815               NaN                         NaN   
10816               NaN                         NaN   
10817               NaN                         NaN   

       Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
10813                         NaN               NaN               NaN  
10814                         NaN               NaN               NaN  
10815                         NaN               NaN               NaN  
10816                         NaN               NaN               NaN  
10817                         NaN               NaN               NaN  

[5 rows x 57 columns]

In [26]:
for key in extracted_data.keys():
    print(f"Key: {key}")
    df = extracted_data[key]
    nunique_series = df.nunique(dropna=False)
    single_unique_cols = nunique_series[nunique_series == 1].index.tolist()
    for col in single_unique_cols:
        unique_val = df[col].unique()
        print(f"  {col}: {unique_val}")

Key: 視線誘導問題.002
  Sensor: ['Eye Tracker']
  Event: [nan]
  Event value: [nan]
  Presented Stimulus name: ['視線誘導問題.002']
  Presented Media name: ['視線誘導問題.002.jpeg']
  Presented Media width: [1920.]
  Presented Media height: [1080.]
  Presented Media position X (DACSpx): [0.]
  Presented Media position Y (DACSpx): [0.]
  Original Media width: [1920.]
  Original Media height: [1080.]
  Mouse position X: [nan]
  Mouse position Y: [nan]
Key: 視線誘導問題.001
  Sensor: ['Eye Tracker']
  Event: [nan]
  Event value: [nan]
  Presented Stimulus name: ['視線誘導問題.001']
  Presented Media name: ['視線誘導問題.001.jpeg']
  Presented Media width: [1920.]
  Presented Media height: [1080.]
  Presented Media position X (DACSpx): [0.]
  Presented Media position Y (DACSpx): [0.]
  Original Media width: [1920.]
  Original Media height: [1080.]
  Mouse position X: [nan]
  Mouse position Y: [nan]
Key: 視線誘導問題.003
  Sensor: ['Eye Tracker']
  Event: [nan]
  Event value: [nan]
  Presented Stimulus name: ['視線誘導問題.003']
  Presen

In [27]:
# 空の辞書に結果を格納
cleaned_data = {}

for key, df in extracted_data.items():
    nunique_series = df.nunique(dropna=False)
    single_unique_cols = nunique_series[nunique_series == 1].index.tolist()

    # そのカラムを除外して新しいデータフレームを作成
    df_cleaned = df.drop(columns=single_unique_cols).copy()

    cleaned_data[key] = df_cleaned

    print(f"{key}: Removed {len(single_unique_cols)} columns -> Remaining {df_cleaned.shape[1]} columns")


視線誘導問題.002: Removed 13 columns -> Remaining 44 columns
視線誘導問題.001: Removed 13 columns -> Remaining 44 columns
視線誘導問題.003: Removed 13 columns -> Remaining 44 columns
視線誘導問題.004: Removed 13 columns -> Remaining 44 columns


### カラムの要素が1つしかない変数の説明 写真1
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-9`|
|Recording name| 録画名|`Recording10`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`17:06:44.023`|
|Recording start time UTC| 録画開始時刻(UTC) |`08:06:44.023`|
|Recording duration| 録画時間|`243039`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`49.3`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`3.61`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`171`pixels|
|Average validation accuracy | 平均バリデーション精度|`13`mm|
|Average validation precision SD | 平均バリデーション予測SD|`17.4`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`18`mm|
|Average validation accuracy |平均バリデーション精度| `1.2`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `1.6`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `1.65`degrees|
|Average validation accuracy |平均バリデーション精度| `45`pixels|
|Average validation precision SD |平均バリデーション予測SD| `61`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `63`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.001`|
|Presented Media name|提示されたメディア名| `視線誘導問題.001.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|

In [28]:
cleaned_data["視線誘導問題.001"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
5136        5136            100915117          2367935840   
5137        5137            100931519          2367952242   
5138        5138            100947999          2367968722   
5139        5139            100965031          2367985754   
5140        5140            100981382          2368002105   
...          ...                  ...                 ...   
7996        7996            148729508          2415750231   
7997        7997            148746149          2415766872   
7998        7998            148763156          2415783879   
7999        7999            148779431          2415800154   
8001        8001            148796072          2415816795   

      Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
5136          2.383190e+09         816.0         659.0                NaN   
5137          2.383206e+09         818.0         670.0                NaN   
5138          2.383223e+09           NaN           NaN                NaN   
5139          2.383240e+09         792.0         628.0              796.0   
5140          2.383256e+09         800.0         597.0              799.0   
...                    ...           ...           ...                ...   
7996          2.431001e+09           NaN           NaN                NaN   
7997          2.431018e+09           NaN           NaN                NaN   
7998          2.431035e+09           NaN           NaN                NaN   
7999          2.431051e+09           NaN           NaN                NaN   
8001          2.431068e+09           NaN           NaN                NaN   

      Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
5136                NaN               816.0               659.0  ...   
5137                NaN               818.0               670.0  ...   
5138                NaN                 NaN                 NaN  ...   
5139              621.0               787.0               634.0  ...   
5140              549.0               802.0               645.0  ...   
...                 ...                 ...                 ...  ...   
7996                NaN                 NaN                 NaN  ...   
7997                NaN                 NaN                 NaN  ...   
7998                NaN                 NaN                 NaN  ...   
7999                NaN                 NaN                 NaN  ...   
8001                NaN                 NaN                 NaN  ...   

      Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
5136                          NaN                        0.4250   
5137                          NaN                        0.4263   
5138                          NaN                           NaN   
5139                       0.5753                        0.4097   
5140                       0.5084                        0.4176   
...                           ...                           ...   
7996                          NaN                           NaN   
7997                          NaN                           NaN   
7998                          NaN                           NaN   
7999                          NaN                           NaN   
8001                          NaN                           NaN   

      Gaze point right Y (MCSnorm)  Eye movement type  \
5136                        0.6100       Unclassified   
5137                        0.6207       Unclassified   
5138                           NaN       EyesNotFound   
5139                        0.5873       Unclassified   
5140                        0.5974       Unclassified   
...                            ...                ...   
7996                           NaN       EyesNotFound   
7997                           NaN       EyesNotFound   
7998                           NaN       EyesNotFound   
7999                           NaN       EyesNotFound   
8001                           NaN       EyesNotFou

In [29]:
cleaned_data["視線誘導問題.001"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-9/user-9-pic-1.csv")

### カラムの要素が1つしかない変数の説明 写真4
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-9`|
|Recording name| 録画名|`Recording10`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`17:06:44.023`|
|Recording start time UTC| 録画開始時刻(UTC) |`08:06:44.023`|
|Recording duration| 録画時間|`243039`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`49.3`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`3.61`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`171`pixels|
|Average validation accuracy | 平均バリデーション精度|`13`mm|
|Average validation precision SD | 平均バリデーション予測SD|`17.4`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`18`mm|
|Average validation accuracy |平均バリデーション精度| `1.2`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `1.6`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `1.65`degrees|
|Average validation accuracy |平均バリデーション精度| `45`pixels|
|Average validation precision SD |平均バリデーション予測SD| `61`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `63`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.004`|
|Presented Media name|提示されたメディア名| `視線誘導問題.004.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|


In [30]:
cleaned_data["視線誘導問題.004"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
10813       10813            196093920          2463114643   
10814       10814            196110732          2463131455   
10815       10815            196127268          2463147991   
10816       10816            196143936          2463164659   
10817       10817            196160655          2463181378   
...           ...                  ...                 ...   
13316       13316            237778084          2504798807   
13317       13317            237794368          2504815091   
13318       13318            237811009          2504831732   
13320       13320            237828007          2504848730   
13321       13321            237844291          2504865014   

       Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
10813          2.478363e+09           NaN           NaN                NaN   
10814          2.478380e+09           NaN           NaN                NaN   
10815          2.478396e+09           NaN           NaN                NaN   
10816          2.478413e+09           NaN           NaN                NaN   
10817          2.478430e+09           NaN           NaN                NaN   
...                     ...           ...           ...                ...   
13316          2.520044e+09           NaN           NaN                NaN   
13317          2.520061e+09           NaN           NaN                NaN   
13318          2.520077e+09           NaN           NaN                NaN   
13320          2.520094e+09           NaN           NaN                NaN   
13321          2.520111e+09           NaN           NaN                NaN   

       Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
10813                NaN                 NaN                 NaN  ...   
10814                NaN                 NaN                 NaN  ...   
10815                NaN                 NaN                 NaN  ...   
10816                NaN                 NaN                 NaN  ...   
10817                NaN                 NaN                 NaN  ...   
...                  ...                 ...                 ...  ...   
13316                NaN                 NaN                 NaN  ...   
13317                NaN                 NaN                 NaN  ...   
13318                NaN                 NaN                 NaN  ...   
13320                NaN                 NaN                 NaN  ...   
13321                NaN                 NaN                 NaN  ...   

       Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
10813                          NaN                           NaN   
10814                          NaN                           NaN   
10815                          NaN                           NaN   
10816                          NaN                           NaN   
10817                          NaN                           NaN   
...                            ...                           ...   
13316                          NaN                           NaN   
13317                          NaN                           NaN   
13318                          NaN                           NaN   
13320                          NaN                           NaN   
13321                          NaN                           NaN   

       Gaze point right Y (MCSnorm)  Eye movement type  \
10813                           NaN       EyesNotFound   
10814                           NaN       EyesNotFound   
10815                           NaN       EyesNotFound   
10816                           NaN       EyesNotFound   
10817                           NaN       EyesNotFound   
...                             ...                ...   
13316                           NaN       EyesNotFound   
13317                           NaN       EyesNotFound   
13318                           NaN       EyesNotFound   
13320                           NaN       EyesNotFo

In [31]:
cleaned_data["視線誘導問題.004"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-9/user-9-pic-4.csv")

### カラムの要素が1つしかない変数の説明 写真2
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-9`|
|Recording name| 録画名|`Recording10`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`17:06:44.023`|
|Recording start time UTC| 録画開始時刻(UTC) |`08:06:44.023`|
|Recording duration| 録画時間|`243039`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`49.3`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`3.61`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`171`pixels|
|Average validation accuracy | 平均バリデーション精度|`13`mm|
|Average validation precision SD | 平均バリデーション予測SD|`17.4`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`18`mm|
|Average validation accuracy |平均バリデーション精度| `1.2`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `1.6`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `1.65`degrees|
|Average validation accuracy |平均バリデーション精度| `45`pixels|
|Average validation precision SD |平均バリデーション予測SD| `61`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `63`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.002`|
|Presented Media name|提示されたメディア名| `視線誘導問題.002.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|

In [32]:
cleaned_data["視線誘導問題.002"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
2105        2105             49757342          2316778065   
2106        2106             49773405          2316794128   
2107        2107             49838223          2316858946   
2108        2108             49869390          2316890113   
2109        2109             49885479          2316906202   
...          ...                  ...                 ...   
4846        4846             95832632          2362853355   
4847        4847             95849319          2362870042   
4848        4848             95866093          2362886816   
4850        4850             95882555          2362903278   
4851        4851             95899278          2362920001   

      Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
2105          2.332035e+09           NaN           NaN                NaN   
2106          2.332051e+09           NaN           NaN                NaN   
2107          2.332116e+09           NaN           NaN                NaN   
2108          2.332147e+09           NaN           NaN                NaN   
2109          2.332163e+09           NaN           NaN                NaN   
...                    ...           ...           ...                ...   
4846          2.378108e+09           NaN           NaN                NaN   
4847          2.378124e+09           NaN           NaN                NaN   
4848          2.378141e+09           NaN           NaN                NaN   
4850          2.378158e+09           NaN           NaN                NaN   
4851          2.378174e+09           NaN           NaN                NaN   

      Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
2105                NaN                 NaN                 NaN  ...   
2106                NaN                 NaN                 NaN  ...   
2107                NaN                 NaN                 NaN  ...   
2108                NaN                 NaN                 NaN  ...   
2109                NaN                 NaN                 NaN  ...   
...                 ...                 ...                 ...  ...   
4846                NaN                 NaN                 NaN  ...   
4847                NaN                 NaN                 NaN  ...   
4848                NaN                 NaN                 NaN  ...   
4850                NaN                 NaN                 NaN  ...   
4851                NaN                 NaN                 NaN  ...   

      Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
2105                          NaN                           NaN   
2106                          NaN                           NaN   
2107                          NaN                           NaN   
2108                          NaN                           NaN   
2109                          NaN                           NaN   
...                           ...                           ...   
4846                          NaN                           NaN   
4847                          NaN                           NaN   
4848                          NaN                           NaN   
4850                          NaN                           NaN   
4851                          NaN                           NaN   

      Gaze point right Y (MCSnorm)  Eye movement type  \
2105                           NaN       EyesNotFound   
2106                           NaN       EyesNotFound   
2107                           NaN       EyesNotFound   
2108                           NaN       EyesNotFound   
2109                           NaN       EyesNotFound   
...                            ...                ...   
4846                           NaN       EyesNotFound   
4847                           NaN       EyesNotFound   
4848                           NaN       EyesNotFound   
4850                           NaN       EyesNotFound   
4851                           NaN       EyesNotFou

In [33]:
cleaned_data["視線誘導問題.002"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-9/user-9-pic-2.csv")

### カラムの要素が1つしかない変数の説明 写真3
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-9`|
|Recording name| 録画名|`Recording10`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`17:06:44.023`|
|Recording start time UTC| 録画開始時刻(UTC) |`08:06:44.023`|
|Recording duration| 録画時間|`243039`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`49.3`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`3.61`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`171`pixels|
|Average validation accuracy | 平均バリデーション精度|`13`mm|
|Average validation precision SD | 平均バリデーション予測SD|`17.4`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`18`mm|
|Average validation accuracy |平均バリデーション精度| `1.2`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `1.6`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `1.65`degrees|
|Average validation accuracy |平均バリデーション精度| `45`pixels|
|Average validation precision SD |平均バリデーション予測SD| `61`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `63`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.003`|
|Presented Media name|提示されたメディア名| `視線誘導問題.003.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|


In [34]:
cleaned_data["視線誘導問題.003"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
8296         8296            153824319          2420845042   
8297         8297            153840868          2420861591   
8298         8298            153857417          2420878140   
8299         8299            153874233          2420894956   
8300         8300            153890786          2420911509   
...           ...                  ...                 ...   
10527       10527            191001565          2458022288   
10528       10528            191018206          2458038929   
10529       10529            191035205          2458055928   
10530       10530            191051488          2458072211   
10532       10532            191068129          2458088852   

       Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
8296           2.436096e+09           NaN           NaN                NaN   
8297           2.436112e+09           NaN           NaN                NaN   
8298           2.436129e+09           NaN           NaN                NaN   
8299           2.436146e+09           NaN           NaN                NaN   
8300           2.436162e+09           NaN           NaN                NaN   
...                     ...           ...           ...                ...   
10527          2.473271e+09           NaN           NaN                NaN   
10528          2.473287e+09           NaN           NaN                NaN   
10529          2.473304e+09           NaN           NaN                NaN   
10530          2.473321e+09           NaN           NaN                NaN   
10532          2.473337e+09           NaN           NaN                NaN   

       Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
8296                 NaN                 NaN                 NaN  ...   
8297                 NaN                 NaN                 NaN  ...   
8298                 NaN                 NaN                 NaN  ...   
8299                 NaN                 NaN                 NaN  ...   
8300                 NaN                 NaN                 NaN  ...   
...                  ...                 ...                 ...  ...   
10527                NaN                 NaN                 NaN  ...   
10528                NaN                 NaN                 NaN  ...   
10529                NaN                 NaN                 NaN  ...   
10530                NaN                 NaN                 NaN  ...   
10532                NaN                 NaN                 NaN  ...   

       Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
8296                           NaN                           NaN   
8297                           NaN                           NaN   
8298                           NaN                           NaN   
8299                           NaN                           NaN   
8300                           NaN                           NaN   
...                            ...                           ...   
10527                          NaN                           NaN   
10528                          NaN                           NaN   
10529                          NaN                           NaN   
10530                          NaN                           NaN   
10532                          NaN                           NaN   

       Gaze point right Y (MCSnorm)  Eye movement type  \
8296                            NaN       EyesNotFound   
8297                            NaN       EyesNotFound   
8298                            NaN       EyesNotFound   
8299                            NaN       EyesNotFound   
8300                            NaN       EyesNotFound   
...                             ...                ...   
10527                           NaN       EyesNotFound   
10528                           NaN       EyesNotFound   
10529                           NaN       EyesNotFound   
10530                           NaN       EyesNotFo

In [35]:
cleaned_data["視線誘導問題.003"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-9/user-9-pic-3.csv")